fleet_adapter_mir_actions/fleet_adapter_mir_actions/rmf_wait_until.py

```
@@ -0,0 +1,483 @@
+                f'will have to be provided in the task description submitted '
+                f'to RMF.'
+            )
+            return
+
+        # Register default signal type
+        default_signal = context.action_config.get('default_signal')
+        if default_signal is not None:
+            if default_signal not in supported_signal_types:
+                raise ValueError(
+                    f'User provided a default signal type {default_signal} '
+                    f'in the action config for WaitUntil MirAction, but the '
+                    f'default signal type is not supported!')
+            elif default_signal not in signals.keys():
+                raise ValueError(
+                    f'User provided a default signal type {default_signal} '
+                    f'in the action config for WaitUntil MirAction, but the '
+                    f'default signal type is not configured!')
+        else:
+            self.context.node.get_logger().warn(
+                f'WaitUntil ActionFactory instantiated for robot '
+                f'[{self.context.name}], but no default signal type has been '
+                f'provided in the action config! If no signal type is '
+                f'populated in the task description, there will be no move '
+                f'off signal available and the robot will wait for the full '
+                f'duration of the timeout when this action is triggered.'
+            )
+
+    def supports_action(self, category: str) -> bool:
+        match category:
+            case 'wait_until':
+                return True
+            case _:
+                return False
+
+    def perform_action(
+        self,
+        category: str,
+        description: dict,
+        execution
+    ) -> MirAction:
+        match category:
+            case 'wait_until':
+                return WaitUntil(
+                    description, execution, self.context,
+                    self.custom_modules)
```

Can be handled with an if case

hmmm, on the other hand, since we expect this function to only return a `MirAction`, if we receive an invalid `category`, should we raise an exception? I would probably prefer to return a `None`, and let upstream handle it more elegantly, but I see that the base class signature has been decided on. With all the initial checks in place, it's highly unlikely that we'd get an invalid category, so if it does happen, it might indicate something very wrong has happened and an exception might be a good thing to raise